gce/gce.go

```diff
@@ -572,6 +572,102 @@ func (s *gceOps) RemoveTags(
 	return err
 }
 
+// SetInstanceGroupSize sets node count for a instance group. Count here is per availability zone
+func (s *gceOps) SetInstanceGroupSize(instanceGroupID, instanceGroupLocation string, count int64, timeout time.Duration) error {
+
+	clusterName, err := s.GetClusterName()
+	if err != nil {
+		return nil
+	}
+
+	nodePoolPath := fmt.Sprintf("projects/%s/locations/%s/clusters/%s/nodePools/%s",
+		s.inst.project, instanceGroupLocation, clusterName, instanceGroupID)
+
+	setSizeRequest := &container.SetNodePoolSizeRequest{
+		Name:      nodePoolPath,
+		NodeCount: count,
+	}
+
+	_, err = s.containerService.Projects.Zones.Clusters.NodePools.SetSize(s.inst.project, instanceGroupLocation, clusterName, instanceGroupID, setSizeRequest).Do()
+	if err != nil {
+		return err
+	}
+
+	if timeout > time.Nanosecond {
+		f := func() (interface{}, bool, error) {
+			clusterInfo, err := s.containerService.Projects.Zones.Clusters.Get(s.inst.project, instanceGroupLocation, clusterName).Do()
+			if err != nil {
+				return nil, true, err
+			}
+
+			if clusterInfo.CurrentNodeCount == count*int64(len(clusterInfo.Locations)) {
+				logrus.Infof("cluster node count of [%s] matches with expected count [%v]. Exiting", clusterName, clusterInfo.CurrentNodeCount)
+				return nil, false, nil
+			}
+
+			return nil,
+				true,
+				fmt.Errorf("cluster node count of [%s] does not match. Expected: [%v], Actual:[%v]. Waiting", clusterName, count*int64(len(clusterInfo.Locations)), clusterInfo.CurrentNodeCount)
```

Comment:
split `fmt.Errorf` into multiple lines to avoid wrapping.

Same for other lines in the review